packages/sources/refund.move

```diff
@@ -0,0 +1,528 @@
+            none()
+        }
+    }
+    
+    public fun get_total_to_refund(pool: &RefundPool): u64 { total_to_refund(&pool.accounting) }
+    public fun get_total_raised(pool: &RefundPool): u64 { total_raised(&pool.accounting) }
+    public fun get_total_claimed(pool: &RefundPool): u64 { total_claimed(&pool.accounting) }
+    public fun get_total_boosted(pool: &RefundPool): u64 { total_boosted(&pool.accounting) }
+    public fun base_funds(pool: &RefundPool): u64 { balance::value(funds(&pool.base_pool)) }
+    public fun booster_funds(pool: &RefundPool): u64 { balance::value(funds(&pool.booster_pool)) }
+    public fun current_liabilities(pool: &RefundPool): u64 { accounting::current_liabilities(&pool.accounting) }
+
+    // === Friends ===
+
+    public(friend) fun uid_mut(pool: &mut RefundPool): &mut UID { &mut pool.id }
+    public(friend) fun unclaimed_mut(pool: &mut RefundPool): &mut Table<address, u64> { &mut pool.unclaimed }
+    public(friend) fun accounting_mut(pool: &mut RefundPool): &mut Accounting { &mut pool.accounting }
+    public(friend) fun booster_pool_mut(pool: &mut RefundPool): &mut Pool { &mut pool.booster_pool }
+
+    public(friend) fun reclaim_funds_(
+        inner_pool: &mut Pool,
+        total_raised: u64,
+        total_unclaimed: u64,
+        ctx: &mut TxContext,
+    ) {
+        let funders = funders_mut(inner_pool);
+        assert!(table::contains(funders, sender(ctx)), EInvalidFunder);
+        let funding_amount = table::remove(funders, sender(ctx));
+
+        let is_last = table::is_empty(funders);
+        let funds = funds_mut(inner_pool);
+        
+        let reclaim_amount = if (is_last) {
+            balance::value(funds)
+        } else {
+            // ReclaimAmount = Leftovers * % Share <=>
+            // ReclaimAmount = Leftovers * FundingAmount/TotalRaised
+            // 
+            // We first upscale then downscale
+            mul_div(funding_amount, total_unclaimed, total_raised)
+
+        };
+
+        let reclaim_funds = coin::from_balance(balance::split(funds, reclaim_amount), ctx);
+        transfer::public_transfer(reclaim_funds, sender(ctx));
+    }
+
+    // === Phase Transitions ===
+    
+    public entry fun start_funding_phase(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        timeout_ts: u64,
+        clock: &Clock,
+    ) {
+        assert_publisher(pub);
+        assert_address_addition_phase(pool);
+        assert!(!table::is_empty(&pool.unclaimed), ERefundPoolHasZeroAddresses);
+        
+        assert!(timeout_ts >= clock::timestamp_ms(clock) + MIN_CLAIM_PERIOD_MS, EInvalidTimeoutTimestamp);
+        option::fill(&mut pool.timeout_ts, timeout_ts);
+
+        next_phase(pool)
+    }
+
+    public entry fun start_claim_phase(
+        pool: &mut RefundPool,
+        clock: &Clock,
+    ) {
+        assert_funding_phase(pool);
+        let timeout_ts = *option::borrow(&pool.timeout_ts);
+        assert_claim_phase_time(timeout_ts, clock);
```

Comment:
I think that when advancing towards claims, we should check the `MIN_CLAIM_PERIOD_MS`, like we do on the funding phase. 

Maybe split into two defaults, one for claims, one for funding periods.

Reply:
Just to confirm if I understood correctly, currently the MIN_CLAIM_PERIOD_MS check  is done when changing from:

funding phase --> claim phase

Do you suggest adding another min_time check from:

add_addresses phase --> funding phase?